ast/substitute/Substitute.cc

```diff
@@ -116,9 +117,26 @@ template <typename T> class SubstWalk {
                     [[maybe_unused]] auto _substituted = subst.substituteSymbolName(name);
                 }
             }
+        } else if (send.fun == core::Names::callWithSplat() || send.fun == core::Names::callWithBlock() ||
+                   send.fun == core::Names::callWithSplatAndBlock() || send.fun == core::Names::checkAndAnd()) {
+            ENFORCE(send.numPosArgs() > 2, "These are special desugar methods, should have at least two args");
+
+            // We're lucky because all of these methods have the symbol they dispatch to as the
+            // positional arg at index 1.
+            auto name = unwrapLiteralToName(send.getPosArg(1));
```

Comment:
Perhaps we should `ENFORCE` that the arg is actually a literal?  I guess it's basically the same thing as `ENFORCE`'ing that the name exists further down, but it is a little more clear if the assertion triggered.

Reply:
Yeah it seemed to me like doing the ENFORCE earlier would have just duplicated the contents of the `unwrapLiteralToName` function. I've added a message to the ENFORCE below to make that hopefully more clear.